packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -11,7 +24,199 @@ declare global {
 /**
  * A nimble styled rich text editor
  */
-export class RichTextEditor extends FoundationElement {}
+export class RichTextEditor extends FoundationElement {
+    /**
+     * @internal
+     */
+    @observable
+    public boldButton!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    @observable
+    public italicsButton!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    @observable
+    public bulletListButton!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    @observable
+    public numberedListButton!: ToggleButton;
+
+    /**
+     * @internal
+     */
+    public editor!: HTMLDivElement;
+
+    private tiptapEditor!: Editor;
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.initializeEditor();
+        this.bindEditorTransactionEvent();
+    }
+
+    /**
+     * @internal
+     */
+    public override disconnectedCallback(): void {
+        super.disconnectedCallback();
+        this.unbindEditorTransactionEvent();
+    }
+
+    /**
+     * Toggle the bold mark and focus back to the editor
+     * @internal
+     */
+    public boldButtonClick(): void {
+        this.tiptapEditor.chain().focus().toggleBold().run();
+    }
+
+    /**
+     * Toggle the bold mark and focus back to the editor
+     * @internal
+     */
+    public boldButtonKeyDown(event: KeyboardEvent): boolean {
```

Comment:
We have an issue with using the `change` event and it is explained in this thread: https://github.com/ni/nimble/pull/1289#discussion_r1227956501

I have updated the rationale for this approach in the PR description.

Reply:
Thanks for the background! You may have already evaluated this approach (I didn't dig too deeply into your research), but one more idea would be to use the toggle buttons' `change` events but call `updateEditorButtonsState()` from the handlers.  In theory that would allow the buttons to get their states from TipTap after the user's attempted  action has been taken into consideration. (You might need to do it asynchronously after tiptap has had a chance to apply the change).

Just an idea! If that doesn't work, I'm ok with the current approach now that it's documented.

Reply:
Tried it in dev tools and looks like the toggle-button behaves poorly and emits a change event on programmatic and interactive change. That may cause an issue trying to listen to the change event.

I like the direction though, make toggle button responsible for its interactions. The rich text editor shouldn't be responsible for what key interactions activate it.

Yea like Jesse said see if the approach described works, if not capture it as tech debt that the toggle buttons should be able to handle their own interactions.

Reply:
Thanks for your suggestions. I tried to use only the `change` event as suggested but it doesn't work as expected. We definitely need to do some more research on the functionality to make it work. Therefore, I created a tech debt and will update the code in future PR. Issue: https://github.com/ni/nimble/issues/1441